src/azul/queues.py

```diff
@@ -220,9 +220,11 @@ def wait_to_stabilize(self) -> int:
         Wait for the work queues to reach a steady state.
         """
         sleep_time = 10
+        timeout_limit = max(config.contribution_lambda_timeout(retry=True),
+                            config.aggregation_lambda_timeout(retry=True))
         queues = self.get_queues(config.work_queue_names)
-        # A generous 10 minutes to accommodate transient stalls
-        total_lengths = deque(maxlen=ceil(10 * 60 / sleep_time))
+        # A generous 10-15 minutes to accommodate transient stalls
```

Comment:
The comment is brittle (the timeout is now derived from configuration) and somewhat misleading (it's a single value and not a range). It would be better to move the comment to the definition of the variable and explain the reasoning behind the new logic.